libraries/render-utils/src/AmbientOcclusionEffect.cpp

```diff
@@ -200,113 +315,16 @@ void AmbientOcclusionEffect::setDepthInfo(float nearZ, float farZ) {
-        current.z = (float)(1.0 / pow((double)radius, RADIUS_POWER));
-    }
-}
-
-void AmbientOcclusionEffect::setLevel(float level) {
-    level = std::max(0.01f, level);
-    if (level != getLevel()) {
-        auto& current = _parametersBuffer.edit<Parameters>().radiusInfo;
-        current.w = level;
-    }
-}
-
-void AmbientOcclusionEffect::setDithering(bool enabled) {
-    if (enabled != isDitheringEnabled()) {
-        auto& current = _parametersBuffer.edit<Parameters>().ditheringInfo;
-        current.x = (float)enabled;
-    }
-}
-
-void AmbientOcclusionEffect::setBordering(bool enabled) {
-    if (enabled != isBorderingEnabled()) {
-        auto& current = _parametersBuffer.edit<Parameters>().ditheringInfo;
-        current.w = (float)enabled;
-    }
-}
-
-void AmbientOcclusionEffect::setFalloffBias(float bias) {
-    bias = std::max(0.0f, std::min(bias, 0.2f));
-    if (bias != getFalloffBias()) {
-        auto& current = _parametersBuffer.edit<Parameters>().ditheringInfo;
-        current.z = (float)bias;
-    }
-}
-
-
-void AmbientOcclusionEffect::setNumSamples(int numSamples) {
-    numSamples = std::max(1.0f, (float) numSamples);
-    if (numSamples != getNumSamples()) {
-        auto& current = _parametersBuffer.edit<Parameters>().sampleInfo;
-        current.x = numSamples;
-        current.y = 1.0f / numSamples;
-    }
-}
-
-void AmbientOcclusionEffect::setNumSpiralTurns(float numTurns) {
-    numTurns = std::max(0.0f, (float)numTurns);
-    if (numTurns != getNumSpiralTurns()) {
-        auto& current = _parametersBuffer.edit<Parameters>().sampleInfo;
-        current.z = numTurns;
-    }
-}
-
-void AmbientOcclusionEffect::setEdgeSharpness(float sharpness) {
-    sharpness = std::max(0.0f, (float)sharpness);
-    if (sharpness != getEdgeSharpness()) {
-        auto& current = _parametersBuffer.edit<Parameters>().blurInfo;
-        current.x = sharpness;
-    }
-}
-
-void AmbientOcclusionEffect::setBlurRadius(int radius) {
-    const int MAX_BLUR_RADIUS = 6;
-    radius = std::max(0, std::min(MAX_BLUR_RADIUS, radius));
-    if (radius != getBlurRadius()) {
-        auto& current = _parametersBuffer.edit<Parameters>().blurInfo;
-        current.y = (float)radius;
-        updateGaussianDistribution();
-    }
-}
-
-void AmbientOcclusionEffect::setBlurDeviation(float deviation) {
-    deviation = std::max(0.0f, deviation);
-    if (deviation != getBlurDeviation()) {
-        auto& current = _parametersBuffer.edit<Parameters>().blurInfo;
-        current.z = deviation;
-        updateGaussianDistribution();
-    }
-}
 void AmbientOcclusionEffect::updateGaussianDistribution() {
     auto coefs = _parametersBuffer.edit<Parameters>()._gaussianCoefs;
     GaussianDistribution::evalSampling(coefs, Parameters::GAUSSIAN_COEFS_LENGTH, getBlurRadius(), getBlurDeviation());
 }
 
 void AmbientOcclusionEffect::run(const render::SceneContextPointer& sceneContext, const render::RenderContextPointer& renderContext) {
```

Comment:
All of these `run` functions just get `RenderArgs*, std::shared_ptr<render::Job::Config>, and Scene*` - can they just be changed to reflect that?